File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
answer = 0


def solve(numbers, idx, res, target):
    if idx == len(numbers):
        if res == target:
            global answer
            answer += 1
        return
    solve(numbers, idx + 1, res + numbers[idx], target)
    solve(numbers, idx + 1, res - numbers[idx], target)

def solution(numbers, target):
    solve(numbers, 0, 0, target)
    global answer
    return answer

File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    int answer = 0;
    public void dfs(int i, int[] numbers, int now, int target){
        if(i == numbers.length){
            if(now == target)
                answer++;
            return;
        }
        dfs(i + 1, numbers, now + numbers[i], target);
        dfs(i + 1, numbers, now - numbers[i], target);
        return;
    }
    public int solution(int[] numbers, int target) {
        dfs(0, numbers, 0, target);
    
        return answer;
    }
}